src/gauging.jl

```diff
@@ -0,0 +1,58 @@
+"""Modify the diagonal elements of an ITensor with a specified function"""
+function modify_diagonal_els_itensor(A::ITensor, f::Function)
+  out = copy(A)
+  is = [[j for i in 1:length(inds(A))] for j in 1:mindim(out)]
+  for i in is
+    out[i...] = f(out[i...])
+  end
+
+  return out
+end
+
+"""Put an ITensorNetwork into the symmetric gauge and also return the message tensors (which are the diagonal bond matrices)"""
+function symmetrise_itensornetwork(ψ::ITensorNetwork)
+  ψψ = ψ ⊗ prime(dag(ψ); sites=[])
+  vertex_groups = nested_graph_leaf_vertices(partition(ψψ, group(v -> v[1], vertices(ψψ))))
+  mts = compute_message_tensors(ψψ; vertex_groups=vertex_groups)
```

Comment:
You can just use `vertex_groups` as a shorthand for `vertex_groups=vertex_groups`.